Src/BlazorApp/Services/RaceService.cs

```diff
@@ -89,14 +91,34 @@ public List<Race> GetRaces()
 
     public List<Race> GetOngoingRaces()
     {
-        var races = _ctx.Races.Where(r => r.EndRace == null && r.StartRace.HasValue).ToList();
+        Expression<Func<Race, bool>> onGoingRacesFilter = race => race.StartRace == null && race.EndRace == null;
+        
+        var races = _ctx.Races.Where(onGoingRacesFilter).ToList();
         return races;
     }
 
     public void AddRace(string name)
     {
-        var newRace = new Race{Name = name};
+        var newRace = new Race {Name = name};
         _ctx.Add(newRace);
         _ctx.SaveChanges();
     }
+
+    public void Delete(int raceId)
+    {
+        var raceAndParticipants = _ctx.Races
+            .Include(r => r.Participants)
+            .FirstOrDefault(r => r.Id == raceId);
+        
+        // check if there are participants in race
+        if (raceAndParticipants?.Participants.Count > 0)
```

Comment:
@terjetopland Program throws an error if user try to delete a race with participants